File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/coupon/adaptor/CouponCampaignAdaptor.java
package band.gosrock.domain.domains.coupon.adaptor;


import band.gosrock.common.annotation.Adaptor;
import band.gosrock.domain.domains.coupon.domain.CouponCampaign;
import band.gosrock.domain.domains.coupon.exception.AlreadyExistCouponCampaignException;
import band.gosrock.domain.domains.coupon.exception.CouponCampaignNotFoundException;
import band.gosrock.domain.domains.coupon.repository.CouponCampaignRepository;
import lombok.RequiredArgsConstructor;

@Adaptor
@RequiredArgsConstructor
public class CouponCampaignAdaptor {
    private final CouponCampaignRepository couponCampaignRepository;

    public CouponCampaign save(CouponCampaign couponCampaign) {
        return couponCampaignRepository.save(couponCampaign);
    }

    public void existsByCouponCode(String couponCode) {
        if (couponCampaignRepository.existsByCouponCode(couponCode))
            throw AlreadyExistCouponCampaignException.EXCEPTION;
    }

    public CouponCampaign findByCouponCode(String couponCode) {
        return couponCampaignRepository
                .findByCouponCode(couponCode)
                .orElseThrow(() -> CouponCampaignNotFoundException.EXCEPTION);
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/coupon/adaptor/CouponCampaignAdaptor.kt
package band.gosrock.domain.domains.coupon.adaptor

import band.gosrock.common.annotation.Adaptor
import band.gosrock.domain.domains.coupon.domain.CouponCampaign
import band.gosrock.domain.domains.coupon.exception.AlreadyExistCouponCampaignException
import band.gosrock.domain.domains.coupon.exception.CouponCampaignNotFoundException
import band.gosrock.domain.domains.coupon.repository.CouponCampaignRepository

@Adaptor
class CouponCampaignAdaptor(
    private val couponCampaignRepository: CouponCampaignRepository,
) {
    fun save(couponCampaign: CouponCampaign): CouponCampaign =
        couponCampaignRepository.save(couponCampaign)

    fun existsByCouponCode(couponCode: String) {
        if (couponCampaignRepository.existsByCouponCode(couponCode)) {
            throw AlreadyExistCouponCampaignException.EXCEPTION
        }
    }

    fun findByCouponCode(couponCode: String): CouponCampaign =
        couponCampaignRepository.findByCouponCode(couponCode)
            .orElseThrow { CouponCampaignNotFoundException.EXCEPTION }
}
